src/ngx-auto-scroll.directive.ts

```diff
@@ -40,6 +40,10 @@ export class NgxAutoScroll implements AfterContentInit, OnDestroy {
         this.scrollDown();
     }
 
+    public isScrolledDown(): boolean {
```

Comment:
This method name lies. Because it returns isLocked property. If someone set `isLocked` to `false` and scroll down manually then he will get wrong state from this method.

Reply:
I changed isLocked to private back, so this should not happen. And anyway, if they scroll, the scroll handler should be invoked and isLocked updated.

Reply:
But method name says `isScrolledDown` and returns `isLocked` so it lies with name.
`isScrolledDown` implementations should look like `return this.nativeElement.scrollTop === this.nativeElement.scrollHeight;`. What do you think?

Reply:
Not exactly, because we should take into account the threshold as well. Do you think it will be better to rename the method to `isLocked`? Then I'll also have to rename the field to `_isLocked`. What do you think?

Reply:
Yes. Method should say what's doing. Now someone can be surprised when asking about isScrolledDown and receive information about isLocked.

Reply:
I renamed the method.